### PR TITLE
Add a 'special offer' variant of the print product option cards for 12for12

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you doing in this PR?

This adds a 'special offer' variant of our `ProductOption` component that adds a border, background, and full width top label, to support the upcoming '12 for £12' promotion for Guardian Weekly.

It also adds some promotion-specific logic to display this component variant, and an amendment to add a link to our terms and conditions below the product price cards.

[**Trello Card**](https://trello.com/c/YONhXPJM)

## Is this an AB test?
- [ ] Yes
- [x] No

## Accessibility test checklist
 - [x] [Tested with screen reader](https://webaim.org/articles/voiceover/)
 - [x] [Navigable with keyboard](https://www.accessibility-developer-guide.com/knowledge/keyboard-only/browsing-websites/)
 - [x] [Colour contrast passed](https://www.whocanuse.com/)

### Mobile, Google Pixel 6, Chrome
Screenshot 2023-05-04 at 15 50 01

### Tablet, iPad 8th gen, Safari
Screenshot 2023-05-04 at 15 50 42

### Desktop, macOS, Firefox
Screenshot 2023-05-04 at 15-47-54 The Guardian Weekly Subscriptions The Guardian

### CODE REVIEW

**Comment by i-hardy** [comment]
[GW 12for12: Guardian Weekly Landing Page changes](https://trello.com/c/YONhXPJM/1149-gw-12for12-guardian-weekly-landing-page-changes)

**Comment by i-hardy** [comment]
[Add guardian book voucher deal in to GW benefits](https://trello.com/c/poHRVZi9/1272-add-guardian-book-voucher-deal-in-to-gw-benefits)

File: support-frontend/assets/pages/weekly-subscription-landing/components/weeklyProductPrices.tsx [hunk 1]
```diff
@@ -90,6 +90,14 @@ const weeklyProductProps = (
 		componentType: 'ACQUISITIONS_BUTTON' as OphanComponentType,
 	};
 
+	// TODO: remove/review this when the 12 for 12 offer is over
+	const isSpecialOffer = promotion?.promoCode === '12for12';
```
Comment (tomrf1) [inline support-frontend/assets/pages/weekly-subscription-landing/components/weeklyProductPrices.tsx:94]: please can this be `promotion?.promoCode.startsWith('12for12')`?
There will be a separate promo for Aus

File: support-frontend/assets/components/product/productOptionStyles.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	from,
+	headline,
+	neutral,
+	space,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+
+export const productOption = css`
+	${textSans.medium()}
```
Comment (marialani) [inline support-frontend/assets/components/product/productOptionStyles.tsx:14]:
```suggestion
	${textSans.medium()};
```

File: support-frontend/assets/pages/weekly-subscription-landing/components/content/benefits.tsx [hunk 3]
```diff
@@ -1,8 +1,37 @@
...
 import BenefitsContainer from './benefitsContainer';
 import BenefitsHeading from './benefitsHeading';
 
-function Benefits(): JSX.Element {
+const coreBenefits = [
+	{
+		content: 'Every issue delivered with up to 35% off the cover price',
+	},
+	{
+		content: "Access to the magazine's digital archive",
```
Comment (marialani) [inline support-frontend/assets/pages/weekly-subscription-landing/components/content/benefits.tsx:11]:
```suggestion
		content: 'Access to the magazine's digital archive',
```

File: support-frontend/assets/pages/weekly-subscription-landing/components/content/prices.tsx [hunk 4]
```diff
@@ -98,6 +104,15 @@ const pricesInfo = css`
 	margin-top: ${space[6]}px;
 `;
 
+const termsLink = css`
+	${textSans.xxsmall()}
```
Comment (marialani) [inline support-frontend/assets/pages/weekly-subscription-landing/components/content/prices.tsx:108]:
```suggestion
	${textSans.xxsmall()};
```

**Review by marialani — COMMENTED** [review]
Spotted some v minor lint issues but code looks good so far

File: support-frontend/assets/pages/weekly-subscription-landing/components/weeklyProductPrices.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/weekly-subscription-landing/components/weeklyProductPrices.tsx]: applied, thanks.

**Pull request merged**